homeassistant/components/switchbot/config_flow.py

```diff
@@ -84,36 +71,58 @@ def async_get_options_flow(
         """Get the options flow for this handler."""
         return SwitchbotOptionsFlowHandler(config_entry)
 
+    def __init__(self):
+        """Initialize the config flow."""
+        self._discovered_devices = {}
+
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle a flow initiated by the user."""
 
-        errors = {}
+        errors: dict[str, str] = {}
 
         if user_input is not None:
-            user_input[CONF_MAC] = user_input[CONF_MAC].replace("-", ":").lower()
+            await self.async_set_unique_id(user_input[CONF_MAC].replace(":", ""))
+            self._abort_if_unique_id_configured()
```

Comment:
It's going to be difficult to test L87. 

It's impossible for duplicate entries to be created as they filtered from user_input in L115